src/main/java/com/artipie/asto/TransactionStorage.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.asto;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Transaction.
+ *
+ * @since 0.10.0
+ */
+public interface TransactionStorage extends Storage {
```

Comment:
@Sammers21 Please add a more detailed JavaDoc. What exactly is a ``Transaction``? 

Also, maybe we can remove the ``Storage`` suffix? Since this interface extends interface ``Storage`` it seems already obvious to me, that it is a Storage Transaction :)

Reply:
> Also, maybe we can remove the Storage suffix? Since this interface extends interface Storage it seems already obvious to me, that it is a Storage Transaction :)

Agree